app/models/manageiq/providers/redfish/physical_infra_manager/event_catcher/runner.rb

```diff
@@ -0,0 +1,38 @@
+require "server_sent_events"
+
+module ManageIQ::Providers::Redfish
+  class PhysicalInfraManager::EventCatcher::Runner \
+      < ManageIQ::Providers::BaseManager::EventCatcher::Runner
+    def monitor_events
+      event_stream.listen do |event|
+        event_monitor_running
+        @queue << event
+      end
+    end
+
+    def stop_event_monitor
+    end
+
+    def queue_event(events)
+      PhysicalInfraManager::EventParser.events_to_hashes(
+        events, @cfg[:ems_id]
+      ).each { |h| EmsEvent.add_queue("add", @cfg[:ems_id], h) }
+    end
+
+    private
+
+    def event_stream
+      @event_stream ||= open_event_stream
+    end
+
+    def open_event_stream
+      address = @ems.with_provider_connection do |connection|
+        connection.EventService&.ServerSentEventUri
+      end
+      raise "Redfish service does not support event streaming" if address.nil?
+
+      $redfish_log.info("Streaming events from #{address}")
+      ServerSentEvents::Client.new(URI(address), ServerSentEvents::Parser.new)
```

Comment:
Its best to add the require here so that we only load gems when this is actually executed, helps keep the global memory usage for the rest of the app down.